src/Microsoft.Framework.PackageManager/Restore/NuGet/NuGetv2Feed.cs

```diff
@@ -170,6 +175,16 @@ public PackageInfo BuildModel(string id, XElement element)
             var idElement = properties.Element(_xnameId);
             var titleElement = element.Element(_xnameTitle);
 
+            var publishElement = properties.Element(_xnamePublish);
+            if (publishElement != null)
+            {
+                DateTime publishDate; 
+                if (DateTime.TryParse(publishElement.Value, out publishDate) && (publishDate == _unlistedPublishedTime))
+                {
+                    return null; 
```

Comment:
If you do what I said above, then you can return the model always and then only select those that are published